src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -1251,4 +1251,691 @@ static class C
             //     public static void M(ref readonly scoped scoped) => throw null;
             Diagnostic(ErrorCode.ERR_FeatureInPreview, "readonly").WithArguments("ref readonly parameters").WithLocation(4, 30));
     }
+
+    [Fact]
+    public void RefReadonlyParameter_ArgumentModifiers()
+    {
+        var source = """
+            class C
+            {
+                static void M(ref readonly int p) => System.Console.Write(p);
+                static void Main()
+                {
+                    int x = 5;
+                    M(x);
+                    M(ref x);
+                    M(in x);
+                }
+            }
+            """;
+        var verifier = CompileAndVerify(source, expectedOutput: "555");
+        verifier.VerifyDiagnostics(
+            // (7,11): warning CS9503: Argument 1 should be passed with 'ref' or 'in' keyword
```

Comment:
>Argument 1 should be passed with 'ref' or 'in' keyword [](http://example.com/codeflow?start=39&length=54)

I think we should use wording that explains what is happening. Do we pass a reference to a copy in this case? If so, we should say that, and then say that, in order to avoid the copy, 'in' or 'ref' should be used.